### DISCUSSION TITLE
Работа при подключении с VPN

### DISCUSSION THREAD

**Discussion by arcanium1** [Anonym-tsk/nfqws-keenetic#26] [Q&A]
Добрый день !
Может ли данный метод работать при подключении к подсети роутера с VPN (у меня внешний ip адрес, на роутере VPN). Т.е. использовать домашний VPN как средство обхода блокировки при подключении с мобильного интернета, например.

**Comment by Riadner** [comment]
Да работает. У меня настроен VPN-сервер IKEv2/IPsec на Keenetic. На мобиле все открывается. А вот на twps не работает.

**Comment by arcanium1** [comment]
Проверил, действительно работает!
Единственное, при автоматическом режиме иногда не открывает некоторые ролики на YouTube. А иногда все идеально. Такое ощущение что на YouTube несколько серверов, которые в авто режиме нужно отдельно по новой прописать

**Reply by Riadner** [reply]
Поэтому я использую только режим list со своим списком сайтов.

**Reply by arcanium1** [reply]
Круто! А можно этот список где-то готовый взять как базу?

**Reply by PavelAnok** [reply]
> Круто! А можно этот список где-то готовый взять как базу?

Список очень большой, оно вам не надо.
Добавь вот эти и все будет работать:

youtube.com
youtu.be
googleapis.com
googlevideo.com
i.ytimg.com
i9.ytimg.com
yt3.ggpht.com
yt3.googleusercontent.com
yt4.ggpht.com
yt4.googleusercontent.com
gvt1.com
gstatic.com
youtubei.googleapis.com
youtube-ui.l.google.com
ytimg.l.google.com
ytstatic.l.google.com

А это для Инстаграм:

cdninstagram.com
gateway.instagram.com
graph.instagram.com
i-fallback.instagram.com
i.instagram.com
instagram.com
instagram.fhrk1-1.fna.fbcdn.net
instagram.fkun2-1.fna.fbcdn.net
instagram.frix7-1.fna.fbcdn.net
instagram.fvno2-1.fna.fbcdn.net
scontent-ams2-1.cdninstagram.com
scontent-ams4-1.cdninstagram.com
scontent.cdninstagram.com
static.cdninstagram.com
www.instagram.com
z-p42-instagram.c10r.instagram.com
cdninstagram.com

**Reply by L61senko** [reply]
А если хочется добавить менее популярный сервис, чем instagram? Например Notion или же там только vpn?

**Reply by Riadner** [reply]
> > Круто! А можно этот список где-то готовый взять как базу?
> 
> Список очень большой, оно вам не надо. Добавь вот эти и все будет работать:
> 
если у вас в списке есть instagram.com. то добавлять *.instagram.com в список уже не надо. То есть у вас в списке сейчас очень много лишнего.

**Reply by pogoodim** [reply]
Скажите, пожалуйста, у меня youtube летает, а все остальное типа facebook или instagram не хочет идти через nfqws. Стоит на keenetic extra. Что не так сделал? 

`# Provider network interface, e.g. eth3
# You can specify multiple interfaces separated by space, e.g. ISP_INTERFACE="eth3 nwg1"
ISP_INTERFACE="ppp0"

# All arguments here: https://github.com/bol-van/zapret (search for `nfqws` on the page)
# HTTP(S) strategy
NFQWS_ARGS="--dpi-desync=fake,split2 --dpi-desync-ttl=0 --dpi-desync-repeats=16 --dpi-desync-split-pos=1 --dpi-desync-fooling=md5sig,badseq --dpi-desync-fake-tls=/opt/etc/nfqws/tls_clienthello.bin"

# QUIC strategy
NFQWS_ARGS_QUIC="--filter-udp=443 --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-repeats=11 --dpi-desync-cutoff=d4 --dpi-desync-fooling=badsum --dpi-desync-fake-quic=/opt/etc/nfqws/quic_initial.bin"

QUIC_PORTS=9993,50000-65535
MODE_FILTER=autohostlist
NFQWS_OPT_DESYNC="--dpi-desync=split2"
NFQWS_OPT_DESYNC_QUIC="--dpi-desync=fake --dpi-desync-repeats=8 --dpi-desync-any-protocol --dpi-desync-autottl=1:1-8"

# UDP strategy (doesn't use lists from NFQWS_EXTRA_ARGS)
NFQWS_ARGS_UDP="--filter-udp=50000-50099 --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-repeats=6 --dpi-desync-cutoff=n2"

# auto - automatically detects blocked resources and adds them to the auto.list
NFQWS_EXTRA_ARGS="--hostlist=/opt/etc/nfqws/user.list --hostlist-auto=/opt/etc/nfqws/auto.list --hostlist-auto-debug=/opt/var/log/nfqws.log --hostlist-exclude=/opt/etc/nfqws/exclude.list"

# list - applies rules only to domains in the user.list
#NFQWS_EXTRA_ARGS="--hostlist=/opt/etc/nfqws/user.list"

# all  - applies rules to all traffic except domains from exclude.list
#NFQWS_EXTRA_ARGS="--hostlist-exclude=/opt/etc/nfqws/exclude.list"

# IPv6 support
IPV6_ENABLED=0

# TCP ports for iptables rules
TCP_PORTS=443

# UDP ports for iptables rules
UDP_PORTS=443,50000:50099

# Syslog logging level (0 - silent, 1 - debug)
LOG_LEVEL=0

NFQUEUE_NUM=200
USER=nobody
CONFIG_VERSION=4
`
